jax/_src/numpy/lax_numpy.py

```diff
@@ -3479,6 +3479,27 @@ def polyadd(a1, a2):
     return a2.at[-a1.shape[0]:].add(a1)
 
 
+@_wraps(np.polyint)
+def polyint(p, m=1, k=None):
+  m = core.concrete_or_error(operator.index, m, "'m' argument of jnp.polyint")
+  p = asarray(p)
+  m = int(m)
+  if m < 0:
+    raise ValueError("Order of integral must be positive (see polyder)")
+  if k is None:
+    k = zeros(m, float)
+  k = atleast_1d(k)
+  if len(k) == 1 and m > 1:
+    k = k[0] * ones(m, float)
+  if len(k) < m:
+    raise ValueError("k must be a scalar or a rank-1 array of length 1 or >m.")
+  if m == 0:
+    return p
+  else:
+    y = concatenate((true_divide(p, arange(len(p), 0, -1)), k[:1]))
+    return polyint(y, m - 1, k=k[1:])
```

Comment:
The tail recursion could be replaced with `lax.scan()`, but I think in practice `m` is always a small integer (I cannot recall having ever used `polyint` with anything except `k=1`) so this is probably faster.

Reply:
I would try to avoid the recursive implementation, if possible. See the similar discussion for the implementation of polyder: https://github.com/google/jax/pull/3403#discussion_r438928282